### ISSUE TITLE
Guidance around using `spk infra generate` secrets from a vault

### ISSUE THREAD

**Issue by NathanielRose** [microsoft/bedrock#868]
**As an:**
Operator

**I want:**
Documented Integration of how to embed Key Vault into the spk [generation pipeline](https://github.com/microsoft/bedrock/issues/827)

**So that:**
Provide guidance on how to leverage KV for handling secrets as environmental variables inside my spk HLD 

**Describe the solution you'd like:**
A section in generation pipeline for spk infra that handles secrets from key vaults as environment variables.

**Acceptance Criteria:**
Guidance that details:
- [x] Deployment of a key vault within the Generation Pipeline with example secrets
- [x] Calling KV keys from a pipeline in task scripts
- [x] Versioning, permissions, and name mapping of keys

**Does this require updates to documentation?:**
Yes

**Comment by andrebriggs** [comment]
@NathanielRose @yradsmikhamthe the updated [epic story](https://github.com/microsoft/bedrock/issues/827) says that you will be using variable group. My assumption was that a AzDO variable group is backed by AzKV. Are you proposing a different approach?

**Comment by yradsmikham** [comment]
@andrebriggs The epic does mention variable groups, and in order to keep tasks independent, we have decided that completing https://github.com/microsoft/bedrock/issues/871 should not be blocked by this spike. For now, we are going to use Variable Groups _not_ backed by KV until we fully understand how to incorporate KV appropriately in the pipeline. When we do, we will iterate.

**Comment by NathanielRose** [comment]
@andrebriggs Thoughts here on adding documentation for this?

**Comment by andrebriggs** [comment]
@NathanielRose @yradsmikham let's chat about this in a meeting before we take it on

**Comment by NathanielRose** [comment]
@andrebriggs & @yradsmikham 
Adding my thoughts for clarity, I believe a good acceptance criteria would be
- Modify the infra generation pipeline to create a keyvault based variable group
- Verify our integration test can reference key vault values
-  Updating our [infra generation doc](https://github.com/CatalystCode/spk/blob/master/guides/infra/spk-infra-generation-pipeline.md) to include a guidance section around versioning and permission handling for keyvault

**Comment by andrebriggs** [comment]
@NathanielRose I think we can be even lower fidelity

I would do this first
>Updating our infra generation doc to include a guidance section around versioning and permission handling for keyvault

We can do the rest over several PRs so the value is is continuous

**Comment by yradsmikham** [comment]
@andrebriggs @NathanielRose 

I feel like it doesn't quite make sense to update documentation if it has not been used or tested.

**Comment by andrebriggs** [comment]
@yradsmikham I thought we have already done this. The point is that we are giving guidance with progress complexity. We are still going to do the script. Remember we are doing this over several PRs
